api/src/api/controllers/brla.controller.ts

```diff
@@ -231,26 +239,62 @@ export const createSubaccount = async (
   res: Response<BrlaEndpoints.CreateSubaccountResponse | BrlaEndpoints.BrlaErrorResponse>,
 ): Promise<void> => {
   try {
-    const { cpf: taxId } = req.body;
+    const { cpf, cnpj, taxIdType } = req.body;
+    
+    const taxId = taxIdType === 'CNPJ' ? cnpj : cpf;
+
+    if (!taxId) {
+      res.status(400).json({ error: 'Missing cpf or cnpj' });
+      return;
+    }
 
     const brlaApiService = BrlaApiService.getInstance();
+    // Convert birthdate from number to BRLA format
+    const birthdate = convertDateToBRLAFormat(req.body.birthdate);
+    // if company startDate field was provided, convert it to BRLA format
+    const startDate = convertDateToBRLAFormat(req.body.startDate);
+
+    let subaccountPayload: RegisterSubaccountPayload = { ...req.body, birthdate, startDate };
+
+    // Extra validation for company fields
+    if (taxIdType === 'CNPJ') {
```

Comment:
Wouldn't we need to check for the business registration that _both_ the CPF and CNPJ fields are provided? The cpf field of the 'partner' is still required when registering a business, isn't it?